pkg/mark/link.go

```diff
@@ -0,0 +1,73 @@
+package mark
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/kovetskiy/mark/pkg/confluence"
+)
+
+type Link struct {
+	MDLink string
+	Link   string
+}
+
+// ResolveRelativeLinks finds links in the markdown, and replaces one pointing
+// to other Markdowns either by own link (if not created to Confluence yet) or
+// witn actual Confluence link
+func ResolveRelativeLinks(
+	api *confluence.API,
+	markdown []byte,
+	base string,
+) (links []Link, collectedErrors error) {
+	re := regexp.MustCompile("\\[.*?\\]\\((.*?)\\)")
```

Comment:
Where links contain anchors (e.g. `../path/to/example.md#second-heading`), this pattern treats that as part of the filename in `ioutil.ReadFile()` below. Moreover, we should preserve these anchors as they're (mostly) compatible between readers.

Consider using `\[[^\]]+\]\(([^\)#]+)?#?(.*)?\)` instead (it's a bit more efficient and will capture both parts of the URL.

```
[example](../path/to/example.md#second-heading)
$1 = "../path/to/example/md"
$2 = "second-heading"

[example](../path/to/example.md)
$1 = "../path/to/example.md
$2 = ""

[example](#heading-in-document)
$1 = ""
$2 = "heading-in-document"
```

Reply:
Then below: 
- when both `$1` and `$2` are present, append `#$2` to the resolved link.
- when `$1` is empty, prevent the protocol prefix and instead use `#$2` verbatim.

Reply:
After some further digging, it appears Confluence doesn't like `#anchor` links alone. Given that it's possible this page hasn't been created yet, we can't deterministically query for its Confluence URL.

I suppose the same is true depending on the order pages are created. (If you create two new pages, A and B, and A links to B, and B links to A, you wouldn't have links working correctly on all pages until after two deployment passes).

I suppose because the tool is designed to run on a single file, this is probably by design, and it puts the responsibility of mapping this on the user.

We could attempt to resolve the _current_ page and link to itself (with anchor) as a fully qualified URL.

Reply:
Thank you, will check the link finding regexp and logic.

Reply:
My implementation handles that "not created page" by self building the URL that page is going to get (works at least with ASCII characters, not sure about UTF8). If page is created already it's URL is fetched from confluence.

With anchor alone I think one way is to get current page's (coming) URL, and append anchor there.

Reply:
Added third group to regexp, as whole content from inside parenthesis is needed when making the actual replace.